File: Algorithms/876. Middle of the Linked List/python/solution.array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: ListNode | None) -> ListNode | None:
        arr: list[ListNode] = [head]
        while arr[-1].next:
            arr.append(arr[-1].next)
        return arr[len(arr) // 2]

File: Algorithms/876. Middle of the Linked List/swift/solution.array.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func middleNode(_ head: ListNode?) -> ListNode? {
        var current: ListNode? = head
        var arr: [ListNode?] = [ListNode]()
        while current != nil {
            arr.append(current!)
            current = current?.next
        }
        return arr[arr.count / 2]
    }
}
